app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/ConferenceTest.kt

```diff
@@ -1,60 +1,80 @@
-            calculateTimeFrame(listOf(opening, closing)) { dateUTC: Long -> Moment.ofEpochMilli(dateUTC).minuteOfDay }
-            assertThat(firstSessionStartsAt).isEqualTo(17 * 60 - 2 * 60) // 17:00h -2h zone offset = 15:00h
-            assertThat(lastSessionEndsAt).isEqualTo(18 * 60 + 30 - 2 * 60) // -> 18:00 -2h zone offset
-        }
+    fun `calculateTimeFrame with frab data spanning multiple days`() {
+        val opening = createSession("Opening", duration = 30, dateUtc = 1536332400000L) // 2018-09-07T17:00:00+02:00
+        val closing = createSession("Closing", duration = 30, dateUtc = 1536504300000L) // 2018-09-09T16:45:00+02:00
+        val (firstSessionStartsAt, lastSessionEndsAt) = createConference(opening, closing)
+        assertThat(firstSessionStartsAt).isEqualTo(17 * 60 - 2 * 60) // 17:00h -2h zone offset = 15:00h
+        assertThat(lastSessionEndsAt).isEqualTo(17 * 60 + 15 - 2 * 60 + Conference.ONE_DAY) // -> 17:15 -2h zone offset + day switch
     }
 
     @Test
-    fun calculateTimeFrameFromPentabarfData() {
-        val opening = Session("Opening").apply {
-            duration = 25
-            relStartTime = 570 // 09:30
-        }
-        val closing = Session("Closing").apply {
-            duration = 10
-            relStartTime = 1070 // 17:50
-        }
-        with(Conference()) {
-            calculateTimeFrame(listOf(opening, closing)) { dateUTC: Long -> Moment.ofEpochMilli(dateUTC).minuteOfDay }
-            assertThat(firstSessionStartsAt).isEqualTo(570) // 09:30
-            assertThat(lastSessionEndsAt).isEqualTo(1080) // 18:00
+    fun `calculateTimeFrame with frab data spanning a single day`() {
+        val opening = createSession("Opening", duration = 30, dateUtc = 1536332400000L) // 2018-09-07T17:00:00+02:00
+        val closing = createSession("Closing", duration = 30, dateUtc = 1536336000000L) // 2018-09-07T18:00:00+02:00
+        val (firstSessionStartsAt, lastSessionEndsAt) = createConference(opening, closing)
+        assertThat(firstSessionStartsAt).isEqualTo(17 * 60 - 2 * 60) // 17:00h -2h zone offset = 15:00h
+        assertThat(lastSessionEndsAt).isEqualTo(18 * 60 + 30 - 2 * 60) // -> 18:30 -2h zone offset
+    }
+
+    @Test
+    fun `calculateTimeFrame with frab data spanning a single day in non-chronological order`() {
+        val opening = createSession("Opening", duration = 30, dateUtc = 1536328800000L) // 2018-09-07T16:00:00+02:00
+        val middle = createSession("Middle", duration = 20, dateUtc = 1536332400000L) // 2018-09-07T17:00:00+02:00
+        val closing = createSession("Closing", duration = 30, dateUtc = 1536336000000L) // 2018-09-07T18:00:00+02:00
+        val (firstSessionStartsAt, lastSessionEndsAt) = createConference(opening, closing, middle)
+        assertThat(firstSessionStartsAt).isEqualTo(16 * 60 - 2 * 60) // 16:00h -2h zone offset = 14:00h
+        assertThat(lastSessionEndsAt).isEqualTo(18 * 60 + 30 - 2 * 60) // -> 18:00 -2h zone offset
```

Comment:
-> 18:30 -2h zone offset

Reply:
:heavy_check_mark: Thank you. Fixed in 7501eec3e6a51e2284fa7eff94c06f9f1916d70b.